examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -0,0 +1,285 @@
+#
+# The number of claims (``ClaimNb``) is a positive integer that can be modeled
+# as a Poisson distribution. It is then assumed to be the number of discrete
+# events occurring with a constant rate in a given time interval
+# (``Exposure``, in units of years). Here we model the frequency
+# ``y = ClaimNb / Exposure``, which is still a (scaled) Poisson distribution,
+# and use ``Exposure`` as `sample_weight`.
+
+df["Frequency"] = df.ClaimNb / df.Exposure
+
+print(
+   pd.cut(df.Frequency, [-1e-6, 1e-6, 1, 2, 3, 4, 5]).value_counts()
+)
+
+print("Average Frequency = {}"
+      .format(np.average(df.Frequency, weights=df.Exposure)))
+
+##############################################################################
+#
+# It worth noting that 96 % of policyholders have zero claims, and if we were
+# to convert this problem into a binary classification task, it would be
+# significantly imbalanced.
+#
+# To evaluate the pertinence of the used metrics, we will consider as a
+# baseline an estimator that returns the mean of the training sample.
+
+df_train, df_test = train_test_split(df, random_state=0)
+
+dummy = make_pipeline(
+    column_trans,
+    DummyRegressor(strategy='mean')
+)
+dummy.fit(df_train, df_train.Frequency,
+          dummyregressor__sample_weight=df_train.Exposure)
+
+##############################################################################
+#
+# The Poisson deviance cannot be computed on negative values predicted by the
+# model, so we set the minimum predicted value to eps,
+
+
+def score_estimator(estimator, df_test, eps=1e-5):
+    """Score an estimatr on the test set"""
+
+    print("MSE: %.3f" % mean_squared_error(
+              df_test.Frequency.values, estimator.predict(df_test),
+              df_test.Exposure.values))
+    print("MAE: %.3f" % mean_absolute_error(
+              df_test.Frequency.values, estimator.predict(df_test),
+              df_test.Exposure.values))
+
+    print("mean Poisson deviance: %.3f" % mean_poisson_deviance(
+            df_test.Frequency.values, np.fmax(estimator.predict(df_test), eps),
+            df_test.Exposure.values))
+
+
+print("DummyRegressor")
+score_estimator(dummy, df_test)
+
+##############################################################################
+#
+# We start by modeling the target variable with the least squares linear
+# regression model,
+
+linregr = make_pipeline(column_trans, LinearRegression())
+linregr.fit(df_train, df_train.Frequency,
+            linearregression__sample_weight=df_train.Exposure)
+
+
+print('Number Negatives: %s / total: %s' % (
+      (linregr.predict(df_train) < 0).sum(),
+      df_train.shape[0]))
+
+print("LinearRegression")
+score_estimator(linregr, df_test)
+
+##############################################################################
+#
+# Next we fit the Poisson regressor on the target variable,
+
+glm_freq = make_pipeline(
+    column_trans,
+    PoissonRegressor(alpha=0, max_iter=1000)
```

Comment:
Maybe set `alpha=1` or smaller since OneHotEncoder is used without `drop=None`.

Reply:
Agreed, maybe we should then use `Ridge` instead of `LinearRegression` as well to be consistent.

Reply:
Maybe this also keeps predictions positive? Lets use comparable alpha. I think in Ridge they scale differently with n_samples.